src/ui/Services/CourseMapper.cs

```diff
@@ -0,0 +1,221 @@
+                Name = CourseDetailsSections.AboutFees,
+                Text = courseEnrichmentModel.FeeDetails});            
+
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.EntryRequirementsQualifications,
+                Text = courseEnrichmentModel.Qualifications});
+
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.EntryRequirementsPersonalQualities,
+                Text = courseEnrichmentModel.PersonalQualities});
+
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.EntryRequirementsOther,
+                Text = courseEnrichmentModel.OtherRequirements});
+                
+            
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.AboutSchools,
+                Text = courseEnrichmentModel.HowSchoolPlacementsWork
+            });
+            
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.AboutTheProvider,
+                Text = orgEnrichmentModel.TrainWithUs
+            });
+            
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection{
+                Name = CourseDetailsSections.AboutTheAccreditingProvider,
+                Text = GetAccreditingProviderEnrichment(ucasCourseData.AccreditingProviderId, orgEnrichmentModel)
+            });
+
+            mappedCourse.DescriptionSections.Add(new CourseDescriptionSection {
+                Name = CourseDetailsSections.TrainWithDisabilities,
+                Text = orgEnrichmentModel.TrainWithDisability
+            });
+
+            return mappedCourse;
+        }
+
+        private string GetAccreditingProviderEnrichment(string accreditingProviderId, InstitutionEnrichmentModel enrichmentModel)
+        {
+            if (string.IsNullOrWhiteSpace(accreditingProviderId))
+            {
+                return "";
+            }
+
+            var enrichment = enrichmentModel.AccreditingProviderEnrichments.FirstOrDefault(x => x.UcasInstitutionCode == accreditingProviderId);
+
+            if (enrichment == null)
+            {
+                return "";
+            }
+
+            return enrichment.Description;
+        }
+
+        private string MapAddress(School school)
+        {
+            var addressFragments =  new List<string>{
+                school.Address1,
+                school.Address2,
+                school.Address3,
+                school.Address4
+            }.Where(x => !string.IsNullOrWhiteSpace(x));
+
+            var postCode = school.PostCode ?? "";
+
+            return addressFragments.Any()
+                ? String.Join(", ", addressFragments) + " " + postCode
```

Comment:
I'm pretty sure you don't need to treat postcode separately, you could just add it to the fragment list.

Minor

Reply:
no, designs say no comma between address lines and postcodes, hence separate

Reply:
oh it's the comma, missed that. retracted!